### PR TITLE
ReClique SSO Integration in openy_gc_auth_reclique_sso module

### PR DESCRIPTION
**Pull request by retif**
**Related Issue/Ticket:**

https://openy.atlassian.net/browse/PRODDEV-139

## Steps to test:

- [ ] Enable openy_gc_auth_reclique_sso module
- [ ] Enter Reclique credentials at Reclique SSO OAuth2 provider settings
- [ ] Enable Reclique SSO OAuth2 provider
- [ ] Try to login at Virtual Y

## Quality checks:

Please check these boxes to confirm this PR covers the following cases:

- Maintaining our upgrade path is essential. Check one or the other:
  - [ ] This PR  provides updates via `hook_update_N` or other means.
  - [ ] No updates are necessary for this change.
- Front end fixes should be tested against all of the Open Y Themes.
  - [ ] Tested against Carnation
  - [ ] Tested against Lily
  - [ ] Tested against Rose
  - [ ] This change does not contain front-end fixes.
- [ ] I have flagged this PR "Needs Review" or pinged the VY devs/QA
  team in Slack

### CODE REVIEW

**Comment by hamrant** [comment]
The current code looks ok for me.

**Review by anpolimus — CHANGES_REQUESTED** [review]
For better naming convention, lets rename it to openy_gc_auth_reclique_sso

File: modules/openy_gc_auth/modules/openy_gc_auth_reclique_oauth2/src/Client.php [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+  /**
+   * Validate user subscription.
+   *
+   * @param object $userData
+   *   User data.
+   *
+   * @return bool
+   *   Returns TRUE if user has active subscription.
+   *
+   * @TODO fix
```
Comment (anpolimus) [inline modules/openy_gc_auth/modules/openy_gc_auth_reclique_oauth2/src/Client.php:167]: If it was fixed, do we need to delete this?

File: modules/openy_gc_auth/modules/openy_gc_auth_reclique_oauth2/src/Plugin/GCIdentityProvider/RecliqueOAuth2.php [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+      $container->get('form_builder')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration(): array {
+    return [
+      'login_mode' => 'present_login_button',
```
Comment (anpolimus) [inline modules/openy_gc_auth/modules/openy_gc_auth_reclique_oauth2/src/Plugin/GCIdentityProvider/RecliqueOAuth2.php:93]: lets add `authorization_server` => `https://[association_slug].recliquecore.com`

**Review by anpolimus — CHANGES_REQUESTED** [review]

**Comment by NadyaRudenko** [comment]
when you log out from Virtual Y, you remain logged in ReClique https://monosnap.com/file/V3g1spPqLyl8dwKkHifD3jnohVwPbu
@retif , @anpolimus

**Comment by retif** [comment]
> when you log out from Virtual Y, you remain logged in ReClique https://monosnap.com/file/V3g1spPqLyl8dwKkHifD3jnohVwPbu
> @retif , @anpolimus

I think that is okay

**Comment by anpolimus** [comment]
@NadyaRudenko Oleks is right.
In the scope of this task, log out from CRM is not included.

**Review by NadyaRudenko — APPROVED** [review]

File: modules/openy_gc_auth/modules/openy_gc_auth_reclique_sso/src/OAuth2Client.php [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+use Drupal\Core\Url;
+use GuzzleHttp\Client as GuzzleHttpClient;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Reclique Oauth2 client wrapper service.
+ *
+ * @package Drupal\openy_gc_auth_reclique_sso
+ */
+class OAuth2Client {
```
Comment (anpolimus) [inline modules/openy_gc_auth/modules/openy_gc_auth_reclique_sso/src/OAuth2Client.php:18]: We also have to rename it to RecliqueClient for better readability

**Review by anpolimus — APPROVED** [review]

**Comment by anpolimus** [comment]
Code is ok.
Tested at build - everything is working properly.

**Comment by TomScarboroughYMCA** [comment]
@NadyaRudenko - is there a way I can securely get a hold of the Client Id and Client Secret needed to test against the Open Y team's staging account with ReClique?

**Comment by NadyaRudenko** [comment]
@anpolimus, please advise ^^^ thanks!

**Comment by TomScarboroughYMCA** [comment]
@anpolimus and @NadyaRudenko - I was able to acquire some test credentials and do my testing, so I'm all-good on that. Hopefully you will see this so you will not get distracted getting me additional test users. Thanks again!

**Pull request merged**